Implementing CSV-Import for persons

nobodyzero1 opened the pull request:
# Description

This pull request adds a button for importing person records via CSV on the "All Persons" page, next to the "Add Person" button. The import makes it easier for users to add multiple people at once, which is especially helpful for onboarding and initial setup.

Importing persons data could significantly speed up the onboarding process, as adding each person manually can be tedious.
To implement the import, I extracted reusable logic from the existing "handle" functions into separate utility functions. The CSV import uses these, and the original handlers could also call these functions to avoid code duplication, but to make the testing of the new code easer I did not do this now.

When the import button is clicked, a dialog opens with instructions and an option to download a CSV template. The template headers are based on the PersonType properties, enrollments, privileges, and assignments. The second row contains translations for each header in the user's language. Assignment columns are generated dynamically, so future changes to assignments require no code changes.

The error handling and messaging are still quite basic at this stage, but in my opinion, the feature is already sufficiently robust for practical use and users would benefit from having it available.

Since I am new to contributing to open source projects like this, I am grateful for any suggestions or corrections. Please let me know if there is anything I can improve or if I should follow a different approach.

Dependencies:
No new dependencies were added.

Fixes # (issue)
No related issue

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

nobodyzero1 commented on src/utils/assignments.ts:
```diff
@@ -0,0 +1,92 @@
+      personAssignments.updatedAt = new Date().toISOString();
+      personAssignments.values = personAssignments.values.filter(
+        (c) => c !== code
+      );
```
this code is already corrected, coderabbit schows old status

nobodyzero1 commented on src/utils/enrollments.ts:
```diff
@@ -0,0 +1,43 @@
+  id: string,
+  value: Date
+) => {
+  if (value === null) return;
```
this code is already corrected, coderabbit schows old status

nobodyzero1 commented on src/utils/person.ts:
```diff
@@ -0,0 +1,74 @@
+    new Date().toISOString();
+};
+
+export const toggleGender = (newPerson: PersonType, value) => {
```
this code is already corrected, coderabbit schows old status

nobodyzero1 commented on src/utils/assignments.ts:
```diff
@@ -0,0 +1,92 @@
+  dataView: string,
+  languageGroups: Array<string>
+): PersonType => {
+  const newPerson = person;
```
onst newPerson = person

nobodyzero1 commented on src/utils/enrollments.ts:
```diff
@@ -0,0 +1,43 @@
+) => {
+  if (value === null) return;
+
+  const current = newPerson.person_data.enrollments.find(
+    (history) => history.id === id
+  );
+
+  current.start_date = formatDate(value, 'yyyy/MM/dd');
+  current.updatedAt = new Date().toISOString();
```
i have already built in this function, code rabbit is showing an old status

nobodyzero1 commented:
Hello!
I’m still quite new to this process, so I’m not sure if there’s anything else I need to change in my code, or if I should just wait for further feedback now.
Could you please let me know how to proceed? Thank you!

ux-git commented:
Hey @nobodyzero1! Thank you so much for your contribution, we do appreciate it!

I already took a look at the deployment and it seems like a great feature. We will review the UX/UI side of the user flow and I'll send you a few comments here in the coming days. It's mostly regarding the button placement, color, and the modal window layout. Small improvements to be clear and easy for users.

And afterwards @rhahao will do the review of the code part. 

Thanks again!

ux-git commented:
@nobodyzero1 sorry for the delay 🙂

**Here's my feedback regarding the user experience flow:**
Overall, the one-window approach is very nice and easy to use, I like it. We can go even one step further, by combining it with the export button. And the good thing is that we already have a similar interface on the Congregation settings page → Import/Export button. Please take a look and feel free to re-use the existing elements and integrate your feature. The structure and elements are identical with a few minor edits:

1. We can **combine the import and export buttons into a single "Import/Export" secondary button** to keep the layout clean:
<img width="312" height="146" alt="Screenshot 2025-07-25 at 21 28 48" src="https://github.com/user-attachments/assets/379b8c0d-0c4b-4878-9d47-3bf166650244" />

2. Once opening the window, we can **keep the import and export content in different tabs**:
<img width="437" height="619" alt="Screenshot 2025-07-25 at 21 36 06" src="https://github.com/user-attachments/assets/3e0bb9c3-f296-4782-8bec-2ddc553b04fb" />
<img width="485" height="702" alt="Screenshot 2025-07-25 at 21 43 07" src="https://github.com/user-attachments/assets/09b1ceff-a3e2-4c1d-8152-b6f176d80858" />

3. Instead of the separate "Download template" button, we can **seamlessly integrate the link** into the description text, while still being eye-catchy enough:
<img width="637" height="154" alt="Screenshot 2025-07-25 at 21 54 14" src="https://github.com/user-attachments/assets/068bc778-ed03-4af1-b8a4-45350f513c5c" />


4. We can have **a separate list of the tips** for filling in the template under the collapsible accordion, making it possible to mention more info there in a structured way:
<img width="562" height="228" alt="Screenshot 2025-07-25 at 21 54 26" src="https://github.com/user-attachments/assets/8a872643-9850-4c53-add1-2e34e7c2897b" />

5. Right before importing, as a last step, would be great to have the selectors of the data that must be imported. Instead of cleaning the columns, this will allow users to quickly adjust what should be imported on-the-go:
<img width="421" height="567" alt="Screenshot 2025-07-25 at 21 54 56" src="https://github.com/user-attachments/assets/52a2e7eb-1f92-4a2c-8604-554de0747749" />

6. After import, let's show the success message to let users know that everything went well:
<img width="1151" height="326" alt="Screenshot 2025-07-25 at 22 00 49" src="https://github.com/user-attachments/assets/1b556475-db7d-4246-8a38-b2921642b2ef" />

7. I created a dedicated icon for the .csv data import, it's available in Figma. Please contact us at support@organized-app.com email to discuss how we can collaborate on that. There's a script to import the icon(s) from the UI Kit.
<img width="110" height="114" alt="Screenshot 2025-07-25 at 22 05 18" src="https://github.com/user-attachments/assets/e2419726-3981-44d8-ab8c-429260a2304f" />

8. Regarding the template:
a) Having only one correct option (for activating the qualification) would make it easier to not mix things up. I’d say having “yes” would be user-friendly instead of more tech-minded "true" or "1". What do you think?
b) Having emergency contacts Field service group (as number 1,2,3) would be also nice to have if possible. But here Haja would suggest better.

ux-git commented:
> @ux-git  Thank you very much for your feedback and the effort you put into this. I’ll do my best to implement the suggested changes as soon as possible. 👍


Thanks so much! But please make sure to contact us via email to get access to Figma designs to use existing styles, icons and design language 🙂

nobodyzero1 commented on src/features/persons/import_export/confirm_import/useConfirmImport.tsx:
```diff
@@ -0,0 +1,233 @@
+        severity: 'error',
+        header: getMessageByCode('error_app_generic-title'),
+        message: getMessageByCode(error.message),
+      });
```
The error is now written only to the console

nobodyzero1 commented on src/features/persons/import_persons/useImportPersons.ts:
```diff
@@ -0,0 +1,513 @@
+        convertValue(enrollmentValue.toLowerCase(), 'date')
+      );
+    }
+  };
```
Code is completely changed, the handlers now are so configured that there should be no problem

nobodyzero1 commented:
> @nobodyzero1 sorry for the delay 🙂
> 
> **Here's my feedback regarding the user experience flow:** Overall, the one-window approach is very nice and easy to use, I like it. We can go even one step further, by combining it with the export button. And the good thing is that we already have a similar interface on the Congregation settings page → Import/Export button. Please take a look and feel free to re-use the existing elements and integrate your feature. The structure and elements are identical with a few minor edits:
> 
>     1. We can **combine the import and export buttons into a single "Import/Export" secondary button** to keep the layout clean:
> 
> 
> <img alt="Screenshot 2025-07-25 at 21 28 48" width="312" height="146" src="https://private-user-images.githubusercontent.com/80993061/470949088-379b8c0d-0c4b-4878-9d47-3bf166650244.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NDkwODgtMzc5YjhjMGQtMGM0Yi00ODc4LTlkNDctM2JmMTY2NjUwMjQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0Y2VlMjgwZDFjZGViOWRhNTkwMzJhN2U1YTI3ZDZmZDJlMTA3MDg3MTE1MmE4ZDk2OTUwZmEzODQxNDliNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qqb6hTij19H6QFlD9Ors0WK1gTb_9mXM7SRi7okaOWM">
> 
>     2. Once opening the window, we can **keep the import and export content in different tabs**:
> 
> 
> <img alt="Screenshot 2025-07-25 at 21 36 06" width="437" height="619" src="https://private-user-images.githubusercontent.com/80993061/470951292-3e0bb9c3-f296-4782-8bec-2ddc553b04fb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NTEyOTItM2UwYmI5YzMtZjI5Ni00NzgyLThiZWMtMmRkYzU1M2IwNGZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExOTVjOTFmY2FmNTRjM2UzZDkwMGNmYmE0OTJlNmZiMjIxOGU5NDMzMjIzOGYxN2MyYzI4YjRkNzI2Zjg5ZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6VJXREIHutf6unjIWbFXkGjZtatwxlq7IG0r--o49rU"> <img alt="Screenshot 2025-07-25 at 21 43 07" width="485" height="702" src="https://private-user-images.githubusercontent.com/80993061/470953852-09b1ceff-a3e2-4c1d-8152-b6f176d80858.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NTM4NTItMDliMWNlZmYtYTNlMi00YzFkLTgxNTItYjZmMTc2ZDgwODU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkODY2YzAxMjBlZTQ4NGZkYWRmODFkZjhlZjNiYTc2MTZjMjcyYzg2ZWNlYjUxYTIwYjE2ZGRjMzE2ODI4N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zdq_Xnw9zDWy4YoqFKVd_kgj3hTm_kHDkFCrmtA_GdI">
> 
>     3. Instead of the separate "Download template" button, we can **seamlessly integrate the link** into the description text, while still being eye-catchy enough:
> 
> 
> <img alt="Screenshot 2025-07-25 at 21 54 14" width="637" height="154" src="https://private-user-images.githubusercontent.com/80993061/470959576-068bc778-ed03-4af1-b8a4-45350f513c5c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NTk1NzYtMDY4YmM3NzgtZWQwMy00YWYxLWI4YTQtNDUzNTBmNTEzYzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmZjhhYmVhMzZiOGNjOTY5ZmVhYzNmZTNhMmI2NGE0NDRiMDM2NDg2NzI0MDRkOTY0ZjJmNmExZDNhMDNjNTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aj33XDa6BR-h7lG50q8DwKpopqCcuTX8k_0nMxi-0YY">
> 
>     4. We can have **a separate list of the tips** for filling in the template under the collapsible accordion, making it possible to mention more info there in a structured way:
> 
> 
> <img alt="Screenshot 2025-07-25 at 21 54 26" width="562" height="228" src="https://private-user-images.githubusercontent.com/80993061/470959684-8a872643-9850-4c53-add1-2e34e7c2897b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NTk2ODQtOGE4NzI2NDMtOTg1MC00YzUzLWFkZDEtMmUzNGU3YzI4OTdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3Mjk3MDQyYjkwMDYxYTFmOTRlNTEwYzNiYzU2NTc3NTMzMjZmMWY3ZDY2MzA1YTM2ODU4ZTliZjUzNDJhOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1FniPoDilv28rLTAvMICmwSiI1tvMIzu3-ujff1koBQ">
> 
>     5. Right before importing, as a last step, would be great to have the selectors of the data that must be imported. Instead of cleaning the columns, this will allow users to quickly adjust what should be imported on-the-go:
> 
> 
> <img alt="Screenshot 2025-07-25 at 21 54 56" width="421" height="567" src="https://private-user-images.githubusercontent.com/80993061/470960021-52a2e7eb-1f92-4a2c-8604-554de0747749.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NjAwMjEtNTJhMmU3ZWItMWY5Mi00YTJjLTg2MDQtNTU0ZGUwNzQ3NzQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiOWMxYmE3NDRiNGQ3MzFiM2VmYTI3YjU1M2EwODIyNjc1ZTBjY2FkNGQwNWMwNWI4ZWNiYmFhZjFhOWNmOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Da4HHzFFUXWoZXuWqQZYdeSSc0k7orGt_BvKrFWGdw8">
> 
>     6. After import, let's show the success message to let users know that everything went well:
> 
> 
> <img alt="Screenshot 2025-07-25 at 22 00 49" width="1151" height="326" src="https://private-user-images.githubusercontent.com/80993061/470962989-1b556475-db7d-4246-8a38-b2921642b2ef.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NjI5ODktMWI1NTY0NzUtZGI3ZC00MjQ2LThhMzgtYjI5MjE2NDJiMmVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwM2I4MjdhMGEyZTg2N2E5ZDJlODI5ODE5NGM3ZGI2NGZhYWExMGM4NDAxOTQzZmI1YTA4MGZmNTgzY2Q1NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BexhoRydorm5xh8D3IGZ8XKjM-8OLsCIFtX-tXuTrLc">
> 
>     7. I created a dedicated icon for the .csv data import, it's available in Figma. Please contact us at [support@organized-app.com](mailto:support@organized-app.com) email to discuss how we can collaborate on that so that you're able to view the Figma designs as well, to make the development easier for you. There's even a script to import the icon(s) from the UI Kit.
> 
> 
> <img alt="Screenshot 2025-07-25 at 22 05 18" width="110" height="114" src="https://private-user-images.githubusercontent.com/80993061/470964547-e2419726-3981-44d8-ab8c-429260a2304f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyOTIxMzcsIm5iZiI6MTc2MTI5MTgzNywicGF0aCI6Ii84MDk5MzA2MS80NzA5NjQ1NDctZTI0MTk3MjYtMzk4MS00NGQ4LWFiOGMtNDI5MjYwYTIzMDRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDA3NDM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzMzEwMDE4ZGZhODJhMzVhZWZhYzg4MTVlNGVmODE5ZWMwMjc1OTA4Y2M4Mzk5M2Q1NWVlZDE1NGRjMmYxZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HmuYdrERHH6dzMo82fXo1h0N6WC38QdTjbW2jzuTUJg">
> 
>     8. Regarding the template:
>        a) Having only one correct option (for activating the qualification) would make it easier to not mix things up. I’d say having “yes” would be user-friendly instead of more tech-minded "true" or "1". What do you think?
>        b) Having emergency contacts Field service group (as number 1,2,3) would be also nice to have if possible. But here Haja would suggest better.
>        c) Perhaps we can include 3-4 rows of dummy data to help users see how is it correct to fill it out? To show where 'yes' is enough, and where some other kind of input is required? Here's the slightly updated template that I think might be a bit more easier for users to understand and fill in correctly:
>        [Template_with_Dates_Contacts_Groups-idea-v4.csv](https://github.com/user-attachments/files/21437627/Template_with_Dates_Contacts_Groups-idea-v4.csv)
> 
>     9. What do you think about adding an ability to import .xlsx file as well? Since the export of the persons record is in this format, so it might be easier to restore a local copy. Or @rhahao maybe you see any better solutions like making the export in .csv too or any other good ideas?
> 
> 
> Let me know your thoughts 🙂
> 
> @rhahao feel free to share your comments as well if you have any at this point

Absolut! Hier ist ein Vorschlag für eine professionelle und freundliche Antwort auf Englisch, die alle deine Punkte klar und strukturiert wiedergibt. Du kannst diesen Text direkt als Kommentar auf GitHub posten.

***

Hi @ux-git,

Thank you so much for your extensive and very helpful feedback from July 25th! I've done my best to implement the points you suggested.

Here is a summary of the changes:

*   **Combined Import/Export Button:** This is done. Import and Export are now consolidated into a single, unified button.
*   **Export Functionality:** To keep the scope of this pull request manageable, I haven't changed the export function itself for now. It works the same way it did before.
*   **Field Selection via Checkboxes:** I thought that if we allow field selection via checkboxes, it would be a good idea to enable this for all fields. To make this easier to manage, I've also added group checkboxes to control multiple fields at once. What do you think about this approach?
*   **Sample Data & Configuration:** Thank you for providing the sample data! I have incorporated it. The examples and the handling of each field during import are now controlled by a configuration array, which should make future extensions much easier.
*   **Expanded Import Capabilities:** It's now also possible to import emergency contacts and field service groups.
*   **Control Values (`yes`, `1`, `true`):** I'd like to propose that we continue to allow not just `yes` but also `1` and `true` as control values for activating options. My reasoning is that CSV imports are often handled by more technically-inclined users who are very comfortable working with `1` or `true`. However, this is just my personal take, and it would be no problem to reduce it to only `yes` if you prefer.
*   **Excel Support:** Expanding the import to include Excel files is a great idea. I would be happy to tackle this in a future pull request, as the changes in this one have already become quite extensive.
*   **Type Change for `first_report`:** I had to allow `null` for the `first_report` date type. The reason is that the base schema for creating new persons pre-fills this property with `null`, and this change was necessary to avoid type conflicts.

I'm looking forward to any further feedback you might have

ux-git commented:
@nobodyzero1 awesome work done! From the UX design perspective it's great, thank you!

Some really minor styling comments after trying the live version:

<img width="680" height="859" alt="Screenshot 2025-11-17 at 22 50 15" src="https://github.com/user-attachments/assets/e37c4515-61ad-4675-8902-90023b05a482" />

1. We don't normally use the underline for text links, simply blue color is fine :) 
2. Could you remove the hover background for the Template filling tips? The whole area must remain clickable as it is, but without the background hover fill, to make the overall composition more simple
3. What do you think about making the tips body text into 'body-small-regular' global text style? This will make the whole block more compact.
4. Could you add the hover fill to the drag-and-drop area as done in #4647? 

_____

@rhahao please share your comments as well. Also, would you want to standartize the import and export formats? Currently as I can see the export is in xslx and import is in csv. Perhaps having both in csv would be more consistent? Or implementing both xslx and csv to be used based on user preference?
